features/config/dao/impl/src/main/java/org/opennms/features/config/dao/impl/util/ValidateUsingConverter.java

```diff
@@ -93,12 +97,20 @@ public ValidateUsingConverter(Class<CONFIG_CLASS> configurationClass)
         }
         this.xsdName = configurationClass.getAnnotation(ValidateUsing.class).value();
         this.rootElement = configurationClass.getAnnotation(XmlRootElement.class).name();
-        this.xmlAccessorType = configurationClass.getAnnotation(XmlAccessorType.class).value();
         this.configurationClass = configurationClass;
         this.validationSchema = this.readXmlSchema();
         this.xmlMapper = new XmlMapper<>(validationSchema.getSchema(), configurationClass);
     }
 
+    public ValidateUsingConverter(final String xsdName,
+                                  final String rootElement)
+            throws IllegalArgumentException, IOException, JAXBException {
+        this.xsdName = Objects.requireNonNull(xsdName);
+        this.rootElement = Objects.requireNonNull(rootElement);
+        this.validationSchema = this.readXmlSchema();
+        this.configurationClass = (Class<CONFIG_CLASS>) ProvisiondConfiguration.class; // TODO: Patrick workaround needs to be removed
```

Comment:
TODO: We need to remove this workaround and remove the need to the JaxB class completely. See here: https://github.com/opennms-forge/notconfd/blob/0d2bdf2cc8bc7d26da591c239128766b399905c5/service/src/main/java/org/opennnms/notconfd/service/xml/XmlMapper.java